File: Mathematics/Sieve_of_Eratosthenes.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

void sieve(int n){
    vector<bool> v(n+1, true);
    // bool v[n] = {true};

    for(int i = 2;i <= n; i++){
        if(v[i]){
            for(int j = 2 * i;j <= n; j+=i){
                v[j] = false;
            }
        }
    }

    for(int i = 2;i <= n;i++){
        if(v[i]){
            cout << i << " ";
        }
    }
}

int main(){
    sieve(10);
}

File: Mathematics/Sieve_of_Eratosthenes.py
def sieve(n):
    v = [True] * (n + 1)

    for i in range(2, n + 1):
        if v[i]:
            for j in range(2 * i, n + 1, i):
                v[j] = False

    for i in range(2, n + 1):
        if v[i]:
            print(i, end=" ")

# Test the function
sieve(10)
